r/R/atac_utils.R

```diff
@@ -459,61 +458,63 @@ write_insertion_bed <- function(fragments, cell_groups,
 #' Columnns are `chr`, `start`, `end`, `cluster`, `name`, `score`, `strand`, `signalValue`, `pValue`, `qValue`, `pointSource`
 #' @details 
 #' File format:
-#'  - Inputs are written such that a bed file, as well as a shell file are written for each cluster.
-#'  - Bed files are written at `<path>/input/<cluster>.bed(.gz)`.
+#'  - Inputs are written such that a bed file used as input into MACS, 
+#' as well as a shell file containing a call to MACS are written for each cluster.
+#'  - Bed files containing `chr`, `start`, and `end` coordinates of insertions are written at `<path>/input/<cluster>.bed(.gz)`.
 #'  - shell commands are written at `<path>/input/<cluster>.sh`.
 #'
-#' Outputs are written to an output directory with a subdirectory for each cluster. 
+#' Outputs are written to an output directory with a subdirectory for each cluster.
 #' Each cluster's output directory contains a file for narrowPeaks, peaks, and summits.
 #'  - NarrowPeaks are written at `<path>/output/<cluster>/<cluster>_peaks.narrowPeak`.
 #'  - Peaks are written at `<path>/output/<cluster>/<cluster>_peaks.xls`.
 #'  - Summits are written at `<path>/output/<cluster>/<cluster>_summits.bed`.
+#'
 #' Only the narrowPeaks file is read into a tibble and returned.
+#' For more information on outputs from MACS, visit the [MACS docs](https://macs3-project.github.io/MACS/docs/callpeak.html)
 #' @inheritParams call_peaks_tile
 #' @export
 call_macs_peaks <- function(fragments, cell_groups, path, effective_genome_size = 2.7e9,
                             insertion_mode = c("start_only", "both", "end_only"),
                             step = c("all", "prep-inputs", "run-macs", "read-outputs"),
-                            macs_version = c("macs3", "macs2"),
-                            additional_params = "--keep-dup all --shift -75 --extsize 150 --nomodel --nolambda",
+                            macs_executable = NULL,
+                            additional_params = "--call-summits --keep-dup all --shift -75 --extsize 150 --nomodel --nolambda",
                             verbose = FALSE,
-                            use_gz = FALSE,
+                            compress_inputs = FALSE,
                             threads = 1) {
   assert_is(fragments, c("IterableFragments", "NULL"))
   assert_is(cell_groups, c("character", "factor"))
   assert_is_numeric(effective_genome_size)
   assert_is_character(path)
-  assert_is(use_gz, "logical")
+  assert_is(compress_inputs, "logical")
   assert_is_wholenumber(threads)
   insertion_mode <- match.arg(insertion_mode)
   step <- match.arg(step)
-  macs_version <- match.arg(macs_version)
   cell_groups <- as.factor(cell_groups)
-  stop_run <- FALSE
 
   # Change cell group spaces into underscores
   levels(cell_groups) <- gsub(" ", "_", levels(cell_groups))
-  if (use_gz) {
+  levels(cell_groups) <- gsub("/", "_", levels(cell_groups))
```

Comment:
I don't think escaping spaces should be needed if our quoting is working properly. I'll write up a little helper for checking for illegal characters like `/` and handling that properly